src/connectors/impls/gcl/writer/meta.rs

```diff
@@ -0,0 +1,604 @@
+                    .as_str()
+                    .unwrap_or("")
+                    .to_string(),
+            });
+        }
+    }
+
+    None
+}
+
+pub(crate) fn default_labels(config: &Config) -> StdHashMap<String, String> {
+    let mut labels = StdHashMap::new();
+
+    // Copy over default label entries
+    if let Some(config_labels) = &config.labels {
+        for (k, v) in config_labels {
+            labels.insert(k.clone(), v.to_string());
+        }
+    }
+
+    labels
+}
+
+pub(crate) fn labels(meta: Option<&HashMap<Cow<str>, Value>>) -> StdHashMap<String, String> {
+    let mut labels = StdHashMap::new();
+
+    // Override with metadata label entries
+    if let Some(has_meta) = meta {
+        if let Some(Value::Object(meta_labels)) = has_meta.get("labels") {
+            for (k, v) in meta_labels.iter() {
+                labels.insert(k.to_string(), v.to_string());
+            }
+        }
+    }
+
+    labels
+}
+
+pub(crate) fn operation(meta: Option<&HashMap<Cow<str>, Value>>) -> Option<LogEntryOperation> {
+    // Override for a specific per event trace
+    if let Some(has_meta) = meta {
+        if let Some(operation @ Value::Object(_)) = has_meta.get("operation") {
+            return Some(LogEntryOperation {
+                id: operation.get("id").as_str().unwrap_or("").to_string(),
+                producer: operation.get("producer").as_str().unwrap_or("").to_string(),
+                first: operation.get_bool("first").unwrap_or(false),
+                last: operation.get_bool("last").unwrap_or(false),
+            });
+        }
+    }
+
+    // Otherwise, None as mapping is optional
+    None
+}
+
+pub(crate) fn trace(meta: Option<&HashMap<Cow<str>, Value>>) -> String {
+    // Override for a specific per event trace
+    if let Some(has_meta) = meta {
+        if let Some(trace) = has_meta.get("trace") {
+            return trace.to_string();
+        }
+    }
+
+    // NOTE user supplied defaults supported at this time
+
+    "".to_string()
+}
+
+pub(crate) fn span_id(meta: Option<&HashMap<Cow<str>, Value>>) -> String {
+    // Override for a specific per event span_id
+    if let Some(has_meta) = meta {
+        if let Some(span_id) = has_meta.get("span_id") {
+            return span_id.to_string();
+        }
+    }
+
+    // NOTE user supplied defaults supported at this time
+
+    "".to_string()
+}
+
+pub(crate) fn trace_sampled(meta: Option<&HashMap<Cow<str>, Value>>) -> Result<bool> {
+    // Override for a specific per event trace_sampled
+    if let Some(has_meta) = meta {
+        if let Some(trace_sampled) = has_meta.get("trace_sampled") {
+            #[allow(clippy::redundant_else)] // Not redundant
+            if let Some(trace_sampled) = trace_sampled.as_bool() {
+                return Ok(trace_sampled);
+            } else {
+                return Err(
+                    "Unable to parse `trace_sampled` from event metadata - expected bool".into(),
+                );
+            }
```

Comment:
The way this is written, the else in fact is redundant it can be simplified w/o the need for an allow:
```suggestion
            return  if let Some(trace_sampled) = trace_sampled.as_bool() {
                Ok(trace_sampled);
            } else {
                Err(
                    "Unable to parse `trace_sampled` from event metadata - expected bool".into(),
                );
            }
```